compiler/src/typed/typecore.re

```diff
@@ -142,138 +139,237 @@ let prim1_type =
+    prim_type(
+      [("int", Builtin_types.type_uint16)],
+      Builtin_types.type_wasmi32,
+    )
+  | Not =>
+    prim_type([("bool", Builtin_types.type_bool)], Builtin_types.type_bool)
   | Box
   | BoxBind => {
       let var = newgenvar(~name="a", ());
-      prim_type([var], Builtin_types.type_box(var));
+      prim_type([("value", var)], Builtin_types.type_box(var));
     }
   | Unbox
   | UnboxBind => {
       let var = newgenvar(~name="a", ());
-      prim_type([Builtin_types.type_box(var)], var);
+      prim_type([("value", Builtin_types.type_box(var))], var);
     }
   | Ignore => {
       let var = newgenvar(~name="a", ());
-      prim_type([var], Builtin_types.type_void);
+      prim_type([("value", var)], Builtin_types.type_void);
     }
   | ArrayLength => {
       let var = newgenvar(~name="a", ());
-      prim_type_labeled(
+      prim_type(
         [("array", Builtin_types.type_array(var))],
         Builtin_types.type_number,
       );
     }
-  | Assert => prim_type([Builtin_types.type_bool], Builtin_types.type_void)
+  | Assert =>
+    prim_type([("bool", Builtin_types.type_bool)], Builtin_types.type_void)
   | Throw =>
-    prim_type([Builtin_types.type_exception], newgenvar(~name="a", ()))
+    prim_type(
+      [("exc", Builtin_types.type_exception)],
+      newgenvar(~name="a", ()),
+    )
   | Magic =>
-    prim_type([newgenvar(~name="a", ())], newgenvar(~name="b", ()))
+    prim_type(
+      [("value", newgenvar(~name="a", ()))],
+      newgenvar(~name="b", ()),
+    )
   | WasmFromGrain =>
-    prim_type([newgenvar(~name="a", ())], Builtin_types.type_wasmi32)
+    prim_type(
+      [("value", newgenvar(~name="a", ()))],
+      Builtin_types.type_wasmi32,
+    )
   | WasmToGrain =>
-    prim_type([Builtin_types.type_wasmi32], newgenvar(~name="a", ()))
+    prim_type(
+      [("value", Builtin_types.type_wasmi32)],
+      newgenvar(~name="a", ()),
+    )
   | WasmUnaryI32({arg_type, ret_type})
   | WasmUnaryI64({arg_type, ret_type})
   | WasmUnaryF32({arg_type, ret_type})
   | WasmUnaryF64({arg_type, ret_type}) =>
     prim_type(
-      [grain_type_of_wasm_prim_type(arg_type)],
+      [("int", grain_type_of_wasm_prim_type(arg_type))],
       grain_type_of_wasm_prim_type(ret_type),
     )
   | WasmMemoryGrow =>
-    prim_type([Builtin_types.type_wasmi32], Builtin_types.type_wasmi32);
+    prim_type(
+      [("size", Builtin_types.type_wasmi32)],
+      Builtin_types.type_wasmi32,
+    );
 
 let prim2_type =
   fun
   | NewRational =>
     prim_type(
-      [Builtin_types.type_wasmi32, Builtin_types.type_wasmi32],
+      [
+        ("numerator", Builtin_types.type_wasmi32),
+        ("denominator", Builtin_types.type_wasmi32),
+      ],
       Builtin_types.type_wasmi32,
     )
   | And
   | Or =>
     prim_type(
-      [Builtin_types.type_bool, Builtin_types.type_bool],
+      [
+        ("value1", Builtin_types.type_bool),
+        ("value2", Builtin_types.type_bool),
```

Comment:
I don't love the numbers in these... Maybe something just like `a` and `b` or `left` and `right`. Open to other suggestions.

Reply:
I think `left` and `right` would be the most elogant. I grabbed the `value1` and `value2` from the current pervasives docs. for now Im going to switch it to the `left` and `right` unless anyone has a better idea.